python/paddle/fluid/dygraph/dygraph_to_static/print_transformer.py

```diff
@@ -39,45 +45,94 @@ def transform(self):
         self.visit(self.root)
 
     # NOTE: deal with print in PY3
-    def visit_Call(self, node):
-        assert isinstance(node, gast.Call)
-        if isinstance(node.func, gast.Name) and node.func.id == 'print':
-            var = self._get_print_var(node)
-            return self._construct_print_node(var)
+    def visit_Assign(self, node):
+        if isinstance(node.value, gast.Call):
+            if isinstance(node.value.func,
+                          gast.Name) and node.value.func.id == 'print':
+                node.value = self._transform_call_node(node.value)
+        return node
+
+    def visit_Expr(self, node):
```

Comment:
Same for bugs like visit_Expr above. For example, an Expr can be 

`a1 and a2 and a3 and print(x)`

BTW, why not write the visit_Call like the old code? if you are handling the `x = print(a)` case, you could get parent node of Call node from static analysis and then handle that. In my opinion it is better than current code.

Reply:
done, thx. for this case, keep `print` is `gast.Call `, do not replace by `gast.Assign`

using static analysis is efficient!